### ISSUE TITLE
Assertion failed (old syntax)

### ISSUE THREAD

**Issue by francoisthire** [rlepigre/lambdapi#78]
On the following example, lambdapi gives an assertion failure 
```
A : Type.

def f : A -> A.

[x] f _ --> x.
```

The error message being 
```
[test.dk, 5:0-13] Uncaught exception [File "src/scope.ml", line 297, characters 11-17: Assertion failed].
```

I don't know if this bug still exists for the new syntax.

**Comment by rlepigre** [comment]
No, this bug cannot happen with the new syntax.

**Comment by rlepigre** [comment]
Fixed in d11e1258a189ad6ba2e79472a4ab46a9d4699a0e. Note that now every context variables must occur in the LHS, otherwise an error is triggered.
